explainaboard/utils/tokenizer.py

```diff
@@ -68,55 +120,51 @@ def detokenize(self, tokens: list[str]) -> str:
         """
         ...
 
-    @abc.abstractmethod
-    def to_dict(self) -> dict:
-        """
-        Return a representation of this class that is serializable in json
-        """
-        ...
+    def serialize(self) -> dict[str, SerializableData]:
+        """See Serializable.serialize."""
+        return {}
 
     @classmethod
-    def from_dict(cls, v: dict) -> Tokenizer:
-        new_v = dict(v)
-        cls_name = new_v.pop('cls_name')
-        thismodule = sys.modules[__name__]
-        return getattr(thismodule, cls_name)(**new_v)
+    def deserialize(cls, data: dict[str, SerializableData]) -> Serializable:
+        """See Serializable.deserialize."""
+        return cls()
+
+
+_tokenizer_registry = TypeRegistry[Serializable]()
 
-    @staticmethod
-    def find_spans(orig_str: str, str_list: list[str]) -> TokenSeq:
-        start = 0
-        ret = TokenSeq()
-        for x in str_list:
-            next_start = orig_str.find(x, start)
-            if next_start == -1:
-                raise ValueError(f'could not "{orig_str}".find({x}, {start})')
-            ret.strs.append(x)
-            ret.positions.append(next_start)
-            start = next_start
-        return ret
 
+def get_tokenizer_serializer() -> PrimitiveSerializer:
+    """Create a serializer for tokenizers.
 
+    Returns:
+        A serializer object for tokenizer classes.
+    """
+    return PrimitiveSerializer(_tokenizer_registry)
+
+
+@final
+@_tokenizer_registry.register("SingleSpaceTokenizer")
 class SingleSpaceTokenizer(Tokenizer):
     """
     Split a string on a single ascii space
     """
 
     @lru_cache(maxsize=20)
     def __call__(self, text: str) -> TokenSeq:
-        ret = TokenSeq()
         start = 0
+        strs: list[str] = []
```

Comment:
Just out of curiosity, I was wondering why this change is made. Is it because the dataclass was turned into `final`? (no changes to the code are necessary)

Reply:
It is because of separation of concerns. This function requires no information from Tokenizer because the arguments provide sufficient information to construct TokenSeq.

Reply:
Maybe it was better to separate this change from this pr to keep the whole change simple.

Reply:
Oh, sorry, my comment was not very clear. I meant: "no changes to this PR are necessary, I'm just asking out of curiosity".

Reply:
Nice, thanks for the clarification!